src/Juvix/FrontendContextualise/ModuleOpen/Environment.hs

```diff
@@ -100,3 +159,189 @@ lookupModMap s =
 removeModMap ::
   HasState "modMap" ModuleMap m => Symbol -> m ()
 removeModMap s = Juvix.Library.modify @"modMap" (Map.delete s)
+
+--------------------------------------------------------------------------------
+-- fully resolve module opens
+--------------------------------------------------------------------------------
+
+-- Precondition ∷ old and new context must be in the same context
+-- also terms must be in either the old map or the new map
```

Comment:
but not both? or every term must be in one or the other?

Reply:
it would be odd if the term is in both at this point.

As we go, we depopulate one map while populating the other....

In reality for this function it doesn't matter.. as long as the symbol lives somewhere we are all good